### PR TITLE
Refactor loading `Runners::Processor` classes

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

This refactoring moves the `Processor` related code from the `CLI` class to the `Processor` class.
I believe this change can reduce such tricky code.

> Link related issues or pull requests.

None.

> Check the following if needed.

- [ ] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: lib/runners/cli.rb [hunk 1]
```diff
@@ -17,18 +17,28 @@ def initialize(argv:, stdout:, stderr:, options_json:)
...
 
-        opts.on("--analyzer=ANALYZER", "Analyzer tool name", all_processor_classes.keys) do |analyzer|
-          @analyzer = analyzer
+        opts.on("--analyzer=ANALYZER", "Analyzer name", Processor.children.keys) do |analyzer|
+          @analyzer = analyzer.to_sym
         end
-      end.parse!(argv)
+      end
 
-      raise OptionParser::MissingArgument.new("--analyzer is required") unless analyzer
```
Comment (ybiquitous) [inline lib/runners/cli.rb:28]: [note] When using `raise`, users view the stack trace too much on their terminal. It is hard to find an important message.

File: lib/runners/cli.rb [hunk 2]
```diff
@@ -17,18 +17,28 @@ def initialize(argv:, stdout:, stderr:, options_json:)
...
+      begin
+        option_parser.parse!(argv)
+      rescue OptionParser::ParseError
+        option_parser.abort
```
Comment (ybiquitous) [inline lib/runners/cli.rb:31]: [note] Use [`OptionParser#abort`](https://ruby-doc.org/stdlib-2.7.2/libdoc/optparse/rdoc/OptionParser.html#abort-method)

**Comment by ybiquitous** [comment]
@yykamei Sorry, could you do a quick review, please?

File: test/cli_test.rb [hunk 3]
```diff
@@ -22,33 +22,13 @@ def options_json(options = { source: new_source })
...
-      assert_equal "invalid argument: --analyzer=FOO", error.message
-    end
-
-    assert_raises OptionParser::MissingArgument do
-      CLI.new(argv: ["--analyzer=rubocop"], stdout: stdout, stderr: stderr, options_json: options_json)
-    end.tap do |error|
-      assert_equal "missing argument: GUID is required", error.message
-    end
-  end
-
```
Comment (ybiquitous) [inline test/cli_test.rb:48]: [note] This test fails due to using `abort`. I think the test case is needless.

**Review by yykamei — APPROVED** [review]
It makes sense.

**Comment by ybiquitous** [comment]
Thank you. 😸

**Pull request merged**